source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -484,6 +490,14 @@ bool CompressorFilterConfig::DirectionConfig::isContentTypeAllowed(
   return true;
 }
 
+bool CompressorFilter::isResponseWithContent(Http::ResponseHeaderMap& headers) const {
+  const auto status = headers.Status();
+  if ((status != nullptr && status->value().getStringView() == "204") || head_request_) {
```

Comment:
Would this cause a behavioral change for proxied 101 responses?     Also I don't think 304 responses can have a body.  I wonder if it's worth refactoring the status code logic in the HTTP/1.1 codec to have a shared utility function.

Either way I think it's worth runtime guarding this feature as it affects the data plane.

Reply:
thanks for the review @alyssawilk . i've added new config option which enables this new behavior (i'm bad with naming though, so more than ready to change it if needed). as for "refactoring the status code logic" could you please elaborate on this? are you talking about some kind of utility like ``` hasContent(status_code) bool ``` which checks internally is it "204/304/something else or not"?

Reply:
Is this function needed? Isn't `end_stream` in `encodeHeaders()` an indication of a header-only response?

Reply:
Yep, that's exactly what I meant with regards to a utility, sorry if that was not clear.   end_stream may also work as a first pass, though I'm not sure if for HTTP/1 it's legal to send a header frame and separate end stream bit for responses without a body.

with regards to guarding, I think rather than having permanent config it'd be fine to just runtime guard - there are instructions in CONTRIBUTING.md  What do you think?

Reply:
sure. will rework w/ runtime guard.